[Added] Lua Strings: .len() (#4783)
Codecademy/docs: content/lua/concepts/strings/terms/len/len.md

---
Title: '.len()'
Description: 'Returns the number of characters that exist in a specific string.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Numbers'
  - 'Functions'
  - 'Strings'
  - 'Characters'
CatalogContent:
  - 'learn-lua'
  - 'paths/computer-science'
---

The **`.len()`** method returns the number of characters that exist in a specific string.

## Syntax

```pseudo
string.len(s)
```

> - `s`: The string to be checked.

Reviewer: ```suggestion
- `s`: The input string whose length is to be determined.
```

> > Note: If an empty string is inserted, the method returns `0`.

Reviewer: The previous one is quite obvious. I believe replacing it with this new warning would be appropriate here.

> ---
> Title: '.len()' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
> Description: 'Returns the number of characters of a string.' # Required; ideally under 150 characters and starts with a present-tense verb (used in search engine results and content previews)
> Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
>   - 'Code Foundations'
> Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
>   - 'Numbers'
>   - 'Functions'
>   - 'Strings'
> CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
>   - 'learn-lua'
>   - 'paths/computer-science'
> ---

Reviewer 2: Modifications:

- Let's remove the comments from the metadata
- Let's change the description for uniqueness
- Let's add another entry to both `Subjects:` and `Tags:`

```suggestion
---
Title: '.len()'
Description: 'Returns the number of characters that exist in a specific string.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Numbers'
  - 'Functions'
  - 'Strings'
  - 'Characters'
CatalogContent:
  - 'learn-lua'
  - 'paths/computer-science'
---
```

> The `.len()` function returns the complete number of characters contained in a string.

Reviewer 2: Let's change the introduction for uniqueness:

```suggestion
The **`.len()`** method returns the number of characters that exist in a specific string.
```

## Syntax

> ```shell
> string.len(s)
> ```
> Replace `s` with your string.
> 
> An empty string `""` would output `0`.

Reviewer 2: Modifications:

- Let's change `shell` to `pseudo` since it's a syntax
- Let's change the sentence construction

````suggestion
```pseudo
string.len(s)
```

- `s`: The string to be checked.

> Note: If an empty string is inserted, the method returns `0`.
````

> ## Example

Reviewer 2: Let's add an introduction to the example:

```suggestion
## Example

The following example demonstrates the usage of the `.len()` method:
```

```lua
town = "Llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch"
lenExample = string.len(town)
print(lenExample)
```

> Output:

Reviewer 2: Let's change the line into a full sentence:

```suggestion
The above code produces the following output:
```

```shell
58
```

> 
> ## Codebyte Example
> 
> ```codebyte/lua
> v1 = "Hi"
> v2 = "Hello!"
> v3 = "How are you doing!?"
> 
> print(string.len(v1))
> print(string.len(v2))
> print(string.len(v3))
> ```

Reviewer 2: Let's remove the codebyte example in Lua as it's not supported yet:
````suggestion

## Codebyte Example

```codebyte/lua
v1 = "Hi"
v2 = "Hello!"
v3 = "How are you doing!?"

print(string.len(v1))
print(string.len(v2))
print(string.len(v3))
```
````
